gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -683,7 +683,8 @@ public static Builder builder() {
   BucketInfo update(BucketInfo bucketInfo, BucketTargetOption... options);
 
   /**
-   * Update blob information.
+   * Update blob information. All the blob's metadata will be replaced using values from the given
+   * {@code blobInfo}. Update will fail if an incomplete {@code blobInfo} is provided.
```

Comment:
> "Update will fail if an incomplete {@code blobInfo} is provided."

I am puzzled, why this is not also required for create (using insert operation) considering [this](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java#L129) and [this](https://cloud.google.com/storage/docs/json_api/v1/objects/insert).

Also, should we indicate what is incomplete (which I think in this case is only content-type)...